### PR TITLE
Adding the tutorial 'Customize Delegate to Run Your Favorite Third Party Tools'

### PR DESCRIPTION
**Pull request by pavanbelagatti**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by KimberlyFields** [comment]
@pavanbelagatti it looks like you may have accidentally committed several files for STO docs. Please review the committed files and remove any files that are not associated with the delegate tutorial.

File: tutorials/orchestrate-security-tests/integrating-snyk-and-owasp-scan-for-nodejs-app.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# Integrating Synk and OWASP Scan for NodeJS Application
```
Comment (kat-enos) [inline tutorials/orchestrate-security-tests/integrating-snyk-and-owasp-scan-for-nodejs-app.md:1]: The doc should be checked for instances of "Synk."

**Review by kat-enos — DISMISSED** [review]
Approving the one tutorial on customizing the tutorial and not the extraneous files that were at first attached to the commit.

**Comment by pavanbelagatti** [comment]
@KimberlyFields @kat-enos Can I please know the status of this?

**Comment by ravilach** [comment]
@pavanbelagatti is this an STO tutorial or a Delegate tutorial?

**Comment by pavanbelagatti** [comment]
I removed the STO related files, and this tutorial should be only related to customising the Delegate article.

**Comment by ravilach** [comment]
> I removed the STO related files, and this tutorial should be only related to customising the Delegate article.

Take a look at the proposed files, the STO items are still there. Can remove those with `git rm` or create a new PR, which ever is easier for you. 

https://github.com/harness/developer-hub/pull/520/files

**Comment by pavanbelagatti** [comment]
@ravilach @KimberlyFields please take a look.

**Comment by ravilach** [comment]
@pavanbelagatti can you add the tile to the /platform tile TSX? Thanks!

**Comment by pavanbelagatti** [comment]
@ravilach Done!

File: src/components/LandingPage/AllTutorials.tsx [hunk 2]
```diff
@@ -429,6 +429,18 @@ const PlatformList: CardItem[] = [
     time: "5 min",
     link: "/tutorials/platform/install-delegate",
   },
+  {
+    title: "Customize the Delegate to Run Your Favorite Third-Party Tools",
```
Comment (kat-enos) [inline src/components/LandingPage/AllTutorials.tsx:433]:
```suggestion
    title: "Customize the delegate to run your favorite third-party tools",

```

File: tutorials/platform/customize-delegate.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip awscliv2.zip
+./aws/install
+```
+
+See the screenshot below for more clarity.
+
+![aws cli step](./static/customizing-delegate/aws_cli_script.png)
+
+Let’s add the shell script as a step just like how we added Git.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:141]:
```suggestion
Let’s add the shell script as a step the same way we added Git.
```

File: tutorials/platform/customize-delegate.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+![specify delegate](./static/customizing-delegate/delegate_advanced_settings.png)
+
+Save everything and run the pipeline. 
+
+![running the pipeline](./static/customizing-delegate/aws_run_pipeline.png)
+You can see the AWS CLI version.
+
+### Install kubectl
+
+Add the following script into your script and apply the yaml just like how we did previously.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:156]:
```suggestion
Add the following script into your script and apply the YAML the same way we did before.
```

File: tutorials/platform/customize-delegate.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+mv kubectl /opt/harness-delegate/custom-client-tools/kubectl
+```
+
+See the screenshot below for more clarity.
+
+![kubectl script image](./static/customizing-delegate/kubectl_init_script.png)
+
+Save the YAML and apply the `kubectl` command for the delegate.
+
+Add the shell script step as shown below with the `kubectl version` command.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:170]:
```suggestion
Add the shell script step as shown below. Use the `kubectl version` command.
```

File: tutorials/platform/customize-delegate.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+![kubectl version adding](./static/customizing-delegate/kubectl_version_command.png)
+
+Make sure you specify the delegate on the **Advanced** tab of the above step.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:174]:
```suggestion
Make sure you specify the delegate on the **Advanced** tab of the step.
```

File: tutorials/platform/customize-delegate.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+```
+curl -O -L  https://releases.hashicorp.com/terraform/0.12.25/terraform_0.12.25_linux_amd64.zip  
+unzip terraform_0.12.25_linux_amd64.zip  
+mv ./terraform /usr/bin/  
+```
+
+Save the YAML and apply the `kubectl` command.
+
+Add the shell script step as shown below, with the `terraform --version` command.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:194]:
```suggestion
Add the shell script step as follows. Use the `terraform --version` command.
```

File: tutorials/platform/customize-delegate.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+You can create everything from one step or create separate steps for each tool. In this tutorial, we created separate steps for each tool.
+
+![separate step](./static/customizing-delegate/seperate_for_each.png)
+
+Save the configuration and run the pipeline. On success, you should see output that includes the different tool version numbers. 
+
+![running the pipeline successfully](./static/customizing-delegate/together_execution.png)
+
+This way, you can install and play with any of your favorite tools on Harness Delegate. This customization gives a lot of flexibility to developers working with these tools.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:253]:
```suggestion
You can use these steps to install and play with any of your favorite tools on Harness Delegate. This customization gives you a lot of flexibility.  
```

File: src/components/LandingPage/AllTutorials.tsx [hunk 9]
```diff
@@ -429,6 +429,18 @@ const PlatformList: CardItem[] = [
     time: "5 min",
     link: "/tutorials/platform/install-delegate",
   },
+  {
+    title: "Customize the Delegate to Run Your Favorite Third-Party Tools",
+    module: "platform",
+    icon: "img/logo.svg",
+    description: (
+      <>Customize the Delegate to run any of your favorite tools Such as Helm, Terraform, AWS CLI, etc.</>
```
Comment (kat-enos) [inline src/components/LandingPage/AllTutorials.tsx:437]:
```suggestion
      <>Customize the delegate to run your favorite tools, including Helm, Terraform, and AWS CLI.</>

```

File: tutorials/platform/customize-delegate.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
+# Customize the Delegate to Run Your Favorite Third-Party Tools
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:1]:
```suggestion
# Customize the delegate to run your favorite third-party tools
```

File: tutorials/platform/customize-delegate.md [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+
+Now that you've got a Harness cloud account, you can create a project.
+
+![about the project](./static/customizing-delegate/about_project.png)
+
+You can start by inviting collaborators to work with you, or you can click **Save and continue**. 
+
+![invite collaborators](./static/customizing-delegate/collaborators.png)
+
+Next, you need to pick a Harness module for your project. Click the **Continuous Delivery** card.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:22]:
```suggestion
Next, pick a Harness module for your project. Click the **Continuous Delivery** card.
```

File: tutorials/platform/customize-delegate.md [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+
+You should see the Git command execution. 
+
+![git command execution](./static/customizing-delegate/git_command_execution.png)
+
+Now, you can use any Git command in your project. 
+
+### Install AWS CLI
+
+You can add the AWS CLI tooling on your Delegate just the way we added Git.
```
Comment (kat-enos) [inline tutorials/platform/customize-delegate.md:127]:
```suggestion
You can add AWS CLI tooling to your delegate the same way we added Git.
```

**Review by kat-enos — DISMISSED** [review]
Approving, but hope you'll commit the few suggested changes. Sorry for the delay in turning this around again. I was out last week on a family matter.

**Comment by pavanbelagatti** [comment]
@kat-enos Done!

**Comment by ravilach** [comment]
Cool,I'll merge this in!

**Pull request merged**